internal/controller/resources/resources.go

```diff
@@ -110,24 +81,41 @@ func deleteResource(ctx context.Context, namespacedName types.NamespacedName, re
 	return nil
 }
 
-// removeFinalizer remove the dana finalizer from the nfspvc object.
-func removeFinalizer(ctx context.Context, nfspvc danaiov1alpha1.NfsPvc, log logr.Logger, k8sClient client.Client) error {
-	controllerutil.RemoveFinalizer(&nfspvc, NfsPvcDeletionFinalizer)
-	if err := k8sClient.Update(ctx, &nfspvc); err != nil {
-		log.Error(err, "unable to remove the finalizer from the NfsPvc - "+nfspvc.Name)
-		return err
+// cleanupNfsPvc cleanup the pvc and the pv that related to the nfspvc.
+func cleanupResources(ctx context.Context, nfspvc danaiov1alpha1.NfsPvc, log logr.Logger, k8sClient client.Client) error {
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := deleteResource(ctx, types.NamespacedName{Name: nfspvc.Name, Namespace: nfspvc.Namespace}, pvc, log, k8sClient); err != nil {
+		return fmt.Errorf("failed to delete pvc - %s: %s", nfspvc.Name, err.Error())
+	}
+
+	pv := &corev1.PersistentVolume{}
+	if err := deleteResource(ctx, types.NamespacedName{Name: nfspvc.Name + "-" + nfspvc.Namespace + "-pv"}, pv, log, k8sClient); err != nil {
+		return fmt.Errorf("failed to delete pv - %s: %s", nfspvc.Name+"-"+nfspvc.Namespace+"-pv", err.Error())
 	}
+
 	return nil
 }
```

Comment:
Comment does not match function name. Also, it's the `resources` package, so it can just be called `cleanup`. Also, handle the failure messages in the client and not here. Only handle each error once. Adding the names to the log does nothing since it's already there from the `WithValues` initiated in the reconciler.